packages/react/src/components/DataTable/DataTable.tsx

```diff
@@ -286,6 +293,10 @@ interface DataTableState<ColTypes extends any[]> {
 }
 
 /**
+ * TODO [@carbon-design-system/monorepo-reviewers]: Is everything in this
+ * comment still accurate? If not, figure out if it should be updated or
+ * deleted.
```

Comment:
Yeah, I think so, but it's a bit convoluted. Trying to parse it out:

* "direct updates to the state" seems like just another way of saying it uses updater functions in setState calls. 
* "normalizing and denormalizing when rendering" is basically the various usages of `reduce`? 
* "accessible through look-up by id" - I assume that's meaning events use `rowId`, `handleOnSelectRow` for example. 
* "updating the state of the single entity will cascade updates to the consumer" aka "we call consumer event handlers on internal state changes"